### PR TITLE
Align Glossary Concepts with OSCAL Model

### PR DESCRIPTION
**Pull request by onselakin**
This PR includes a few updates to the glossary to ensure alignment with the OSCAL model's definitions and terminology. The changes made are as follows:

The term "Assessment" has been redefined to match the OSCAL's terminology. The specific activities related to these are now referred to as "Assessment Actions." The term AttestationProvider has been replaced with "Assessment Action" accordingly.

A new concept, "Observation," has been added to glossary to align with OSCAL. Observations represent the findings or results from Assessment Actions. 

Please provide your insights, comments, or any suggestions for improvement.

EDIT: I've also removed the sub-component definitions that fall under Runtime and Core Services to make the glossary lighter, as they all have definitions and diagrams presented on their relevant pages.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -5,53 +5,38 @@ sidebar_label: Glossary
...
-| SMF24 | |
-| Senior Management Function 24 | the person in an organisation held responsible for operational failings. |
-| Variance | |
-
+| Name                  | Definition                                                                                                                                                                                                                               |
+|-----------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Assessment            | The overall process of evaluating a specific system, control, or process. It includes planning, executing various tests or checks, and drawing conclusions about compliance, performance, or quality.                                    |
+| Attestation           | A formal declaration or statement that confirms that something is true, based on the Assessment and its Observations.                                                                                                                    |
```
Comment (robel-yemane) [inline docs/glossary.md]: `Attestation           | A formal declaration or statement confirming the truth of something, based on the assessment and its observations.`

**Review by robel-yemane — COMMENTED** [review]
nit

**Comment by gusfcarvalho** [comment]
Shouldn't we add Findings and Risk as part of this glossary as well? :)

**Comment by onselakin** [comment]
> Shouldn't we add Findings and Risk as part of this glossary as well? :)

Is Findings different from Observations @gusfcarvalho?

**Comment by gusfcarvalho** [comment]
from OSCAL source at least, they are two individual things: . I'm not sure what are they different though, but it's better to model it out:https://pages.nist.gov/OSCAL/concepts/layer/assessment/assessment-results/

from the POA&M, findings seem to be "general remarks", observations are "important remarks". But that's just a guess

**Comment by onselakin** [comment]
Looks like they are related but not the same:

Observation: This term usually refers to the empirical data or information gathered during an assessment or auditing process. Observations might include detected vulnerabilities, non-compliance with policies, or other elements related to security controls and their implementation. These details are often used to inform a broader analysis of risk.

Finding: A finding refers to a conclusion or determination based on the observations made during an assessment. Findings can identify gaps, weaknesses, or failures in controls, and may also point to areas of strength and compliance. Findings are essentially the analyzed results of the observations, which can further inform the risk profile of the organization.

Adding them to the glossary.

**Review by robel-yemane — APPROVED** [review]

**Review by gusfcarvalho — APPROVED** [review]

**Comment by gusfcarvalho** [comment]
I think I need to add Robel to the approvers 🤔

**Pull request merged**